src/uj_arch.h

```diff
@@ -9,8 +9,22 @@
 #ifndef _UJ_ARCH_H
 #define _UJ_ARCH_H
 
-/* Select native OS if no target OS defined. */
+/* Target OS. Values match LuaJIT's definitions. */
+#define UJIT_OS_LINUX 2
+#define UJIT_OS_OSX 3
+
+#ifndef UJIT_OS
+#error "UJIT_OS is expected to be set by the build system"
+#endif /* !UJIT_OS */
+
+#if UJIT_OS == UJIT_OS_LINUX
 #define UJ_OS_NAME "Linux"
+#elif UJIT_OS == UJIT_OS_OSX
+#define UJ_OS_NAME "OSX"
+#endif
+
+#define UJ_TARGET_LINUX (UJIT_OS == UJIT_OS_LINUX)
+#define UJ_TARGET_OSX (UJIT_OS == UJIT_OS_OSX)
```

Comment:
In my opinion, this is a bit cumbersome... Can't we just define UJIT_OS_LINUX or UJIT_OS_OSX in CMake and then write code like this?

```c
#ifdef UJIT_OS_LINUX
#define UJ_OS_NAME "Linux"
#endif 

#ifdef  UJIT_OS_OSX
#define UJ_OS_NAME "OSX"
#endif 

#if !defined(UJIT_OS_LINUX) && !defined(UJIT_OS_OSX)
#error "..."
#endif 
```

This is closer to how this code usually gets written:

```c
#ifdef __linux__ 
#define UJ_OS_NAME "Linux"
#else 
...
```

Reply:
Also we can later use UJIT_OS_LINUX instead of UJIT_TARGET_LINUX (same for Mac)

Reply:
Simplified defines. Thanks!